docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,601 @@
+
+## Start Coding
+
+Let’s create the application!
+1. Create a new Python file by clicking on **File** > **New File…** (`kb(workbench.action.files.newFile)`), and then select "Python File".
+2. Save it as `main.py` (`kb(workbench.action.files.saveAs)`) in the `groceries-plugin` folder.
+3. Add the following code to `main.py` and save the file:
+    ```python
+    from fastapi import FastAPI
+
+    app = FastAPI()
+
+    @app.get("/")
+    def root():
+        return {"message": "Hello World"}
+    ```
+4. Run the code by starting up the debugger (`kb(workbench.action.debug.start)`)
+5. From the drop-down menu, select the **FastAPI** configuration option from the drop-down list:
+![Drop down with debugger configuration options, with FastAPI being highlighted.](images/fastapi-tutorial/fastapi_debug_config_option.png)
+
+    This will invoke uvicorn to start the application server through the debugger, which will allow us to step through the code to inspect its behavior. You will see something like the following in the terminal:
+
+    ![Uvicorn server running message displayed in the terminal, with an URL to access the app.](images/fastapi-tutorial/fastapi_debug_terminal.png)
+
+6. `Ctrl` + click the http://127.0.0.1:8000/ URL in the terminal to open your default browser to that address:
+![Hello World message displayed in the browser.](images/fastapi-tutorial/helloworld_browser.png)
+7. Stop the debugger by clicking on the stop icon in the debug toolbar, or through `kb(workbench.action.debug.stop)`.
+
+## Create a model for grocery list items
+Now that we have the FastAPI app working, let’s define our grocery list items by creating a Pydantic model.
+Pydantic is a data validation and parsing library that integrates seamlessly with FastAPI. It allows you to define data models using Python classes with [type hints](https://docs.python.org/3/library/typing.html) for automatic validation and parsing of incoming data (which are called payloads) in API requests.
+
+Pylance, the default language server for Python in VS Code, supports type hinting features that can be helpful for working with Pydantic models and FastAPI. Let's enable a few of them:
+1. Open the Settings UI page (`kb(workbench.action.openSettings)`)
+2. Search for "python type checking mode" and set it to "basic" to enable basic type checking. This will enable Pylance to show diagnostics and warnings to catch simple type-related errors. Alternetivaly, you can set it to "strict" to enable more advanced [type checking rules](https://microsoft.github.io/pyright/#/configuration?id=diagnostic-rule-defaults).
+    ![Python Analysis Type Checking Mode options (off, basic and strict) in settings UI page.](images/fastapi-tutorial/type_checking_mode_setting.png)
+3. Next, search for "Python inlay type hints", and enable inlay hints for Variable Types and Function Return Types:
```

Comment:
The inlay hints part sounds to me like an optional part.
I suggest to either have it at the end of the document or to mention this is optional goodness.
In the tutorial I wold like to have the minimal steps to get into a good state, and then additional I can do more stuff to make my life easier.

On a related note, why don't we turn these on by default?

Reply:
That's a good point. I made some edits to make it clear this is optional, but also added a bit more context around the value of using Pylance's typing features with FastAPI projects. 

We have plans to turn the inlay hints on by default for files we see explicit type information cause we agree it'd be helpful. However we don't have plans to add it by default in all Python files given a lot of people in the Python community would be highly annoyed by that 😅 Typing in Python in usually seen as essential for production-ready code, but for data analysis, prototyping and basic scripting, a lot of folks in the community find types to be too constrictive and even distracting.